hls4ml/templates/vivado/nnet_utils/nnet_activation_stream.h

```diff
@@ -341,6 +341,309 @@ void softmax(hls::stream<data_T> &data, hls::stream<res_T> &res){
+//       Softplus Activation
+// *************************************************
+
+template<class data_T, class res_T, typename CONFIG_T>
+void softplus(hls::stream<data_T> &data, hls::stream<res_T> &res) {
+    // Initialize the lookup table
+#ifdef __HLS_SYN__
+    bool initialized = false;
+    typename CONFIG_T::table_t softplus_table[CONFIG_T::table_size];
+#else
+    static bool initialized = false;
+    static typename CONFIG_T::table_t softplus_table[CONFIG_T::table_size];
 #endif
+    if (!initialized) {
+        init_softplus_table<CONFIG_T, CONFIG_T::table_size>(softplus_table);
+        initialized = true;
+    }
+
+    SoftplusActLoop: for (int i = 0; i < CONFIG_T::n_in / res_T::size; i++) {
+        #pragma HLS PIPELINE
+
+        data_T in_data = data.read();
+        res_T out_data;
+        #pragma HLS DATA_PACK variable=out_data
+
+        SoftplusPackLoop: for (int j = 0; j < res_T::size; j++) {
+            #pragma HLS UNROLL
+            int data_round = in_data[j]*CONFIG_T::table_size/16;
+            int index = data_round + 8*CONFIG_T::table_size/16;
+            if (index < 0)   index = 0;
+            else if (index > CONFIG_T::table_size-1) index = CONFIG_T::table_size-1;
+            out_data[j] = softplus_table[index];
+        }
+        res.write(out_data);
+    }
+}
+
+
+// *************************************************
+//       Softsign Activation
+// *************************************************
+
+template<class data_T, class res_T, typename CONFIG_T>
+void softsign(hls::stream<data_T> &data, hls::stream<res_T> &res) {
+    // Initialize the lookup table
+#ifdef __HLS_SYN__
+    bool initialized = false;
+    typename CONFIG_T::table_t softsign_table[CONFIG_T::table_size];
+#else
+    static bool initialized = false;
+    static typename CONFIG_T::table_t softsign_table[CONFIG_T::table_size];
+#endif
+    if (!initialized) {
+        init_softsign_table<CONFIG_T, CONFIG_T::table_size>(softsign_table);
+        initialized = true;
+    }
+
+    SoftsignActLoop: for (int i = 0; i < CONFIG_T::n_in / res_T::size; i++) {
+        #pragma HLS PIPELINE
+
+        data_T in_data = data.read();
+        res_T out_data;
+        #pragma HLS DATA_PACK variable=out_data
+
+        SoftsignPackLoop: for (int j = 0; j < res_T::size; j++) {
+            #pragma HLS UNROLL
+            int data_round = in_data[j]*CONFIG_T::table_size/16;
+            int index = data_round + 8*CONFIG_T::table_size/16;
+            if (index < 0)   index = 0;
+            else if (index > CONFIG_T::table_size-1) index = CONFIG_T::table_size-1;
+            out_data[j] = softsign_table[index];
+        }
+        res.write(out_data);
+    }
+}
+
+
+// *************************************************
+//       ELU Activation
+// *************************************************
+template<class data_T, class res_T, typename CONFIG_T>
+void elu(hls::stream<data_T> &data, typename data_T::value_type alpha, hls::stream<res_T> &res) {
```

Comment:
It is possible to parse the ELU activation without `alpha` parameter. In that case it should default to `1.0`. We handle this [case](https://github.com/fastmachinelearning/hls4ml/blob/master/hls4ml/templates/vivado/nnet_utils/nnet_activation.h#L718) in the other implementation, can you add it here as well?

Reply:
@vloncar  Nice catch, I totally missed that function definition while reading through nnet_activation.h. Just added it now with the latest commit.